src/d3a/models/market/market_structures.py

```diff
@@ -263,46 +255,56 @@ class Trade:
     seller_id: str = None
     buyer_id: str = None
 
-    def __str__(self):
+    def __str__(self) -> str:
         return (
             "{{{s.id!s:.6s}}} [origin: {s.seller_origin} -> {s.buyer_origin}] "
-            "[{s.seller} -> {s.buyer}] {s.offer.energy} kWh @ {s.offer.price} {rate} "
-            "{s.offer.id} [fee: {s.fee_price} cts.]".
-            format(s=self, rate=round(self.offer.energy_rate, 8))
+            "[{s.seller} -> {s.buyer}] {s.offer_bid.energy} kWh @ {s.offer_bid.price} {rate} "
+            "{s.offer_bid.id} [fee: {s.fee_price} cts.]".
+            format(s=self, rate=round(self.offer_bid.energy_rate, 8))
         )
 
     @classmethod
-    def _csv_fields(cls):
+    def _csv_fields(cls) -> Tuple:
         return (tuple(cls.__dataclass_fields__.keys())[1:2] + ("rate [ct./kWh]", "energy [kWh]") +
                 tuple(cls.__dataclass_fields__.keys())[3:5])
 
-    def _to_csv(self):
-        rate = round(self.offer.energy_rate, 4)
+    def _to_csv(self) -> Tuple:
+        rate = round(self.offer_bid.energy_rate, 4)
         return (tuple(asdict(self).values())[1:2] +
-                (rate, self.offer.energy) +
+                (rate, self.offer_bid.energy) +
                 tuple(asdict(self).values())[3:5])
 
-    def to_json_string(self):
+    def to_json_string(self) -> str:
         # __dict__ instead of asdict to not recursively deserialize objects
         trade_dict = deepcopy(self.__dict__)
-        trade_dict["offer"] = trade_dict["offer"].to_json_string()
+        trade_dict["offer_bid"] = trade_dict["offer_bid"].to_json_string()
         if key_in_dict_and_not_none(trade_dict, "residual"):
             trade_dict["residual"] = trade_dict["residual"].to_json_string()
         if key_in_dict_and_not_none(trade_dict, "offer_bid_trade_info"):
             trade_dict["offer_bid_trade_info"] = (
                 trade_dict["offer_bid_trade_info"].to_json_string())
         return json.dumps(trade_dict, default=my_converter)
 
-    def serializable_dict(self):
+    @property
+    def is_bid_trade(self) -> bool:
```

Comment:
Nit: a possible alternative would just be
```suggestion
    def is_bid(self) -> bool:
```
Do you think this would be accurate? (Same for the offer part)

Reply:
No, IMO this would be confusing. A trade can not be a bid. This is to check if the trade was issued when accepting a bid. Please leave the property name as is, thanks!